pkgs/dartpad_ui/lib/model/model.dart

```diff
@@ -482,34 +508,82 @@ class AppServices {
   Future<FormatResponse> format(SourceRequest request) async {
     try {
       appModel.formattingBusy.value = true;
-      return await services.format(request);
+      if (useWebsockets) {
```

Comment:
There's a lot of `useWebsockets` checks in this class. I think it would be better to create a common interface (DartPadService) and have HttpService and WebSocketService implement that. That way, the AppServices class can then create the one it needs based on the server configuration.